src/coreclr/vm/codeman.cpp

```diff
@@ -5756,6 +5756,84 @@ int NativeUnwindInfoLookupTable::LookupUnwindInfoForMethod(DWORD RelativePc,
     return -1;
 }
 
+int HotColdMappingLookupTable::LookupMappingForMethod(ReadyToRunInfo* pInfo, ULONG MethodIndex)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+        SUPPORTS_DAC;
+    } CONTRACTL_END;
+
+    if (pInfo->m_nScratch == 0)
+    {
+        return -1;
+    }
+
+    // Casting the lookup table's size to an int is safe:
+    // We index the RUNTIME_FUNCTION table with ints, and the lookup table
+    // contains a subset of the indices in the RUNTIME_FUNCTION table.
+    // Thus, |lookup table| <= |RUNTIME_FUNCTION table|.
+    _ASSERTE(pInfo->m_nScratch <= pInfo->m_nRuntimeFunctions);
```

Comment:
While we are at it, we might as well `_ASSERTE(pInfo->m_nScratch % 2 == 0)`

Reply:
Gotcha, just pushed a revision.